Mathlib/FieldTheory/Finite/Polynomial.lean

```diff
@@ -179,12 +178,14 @@ variable [CommRing K]
 
 noncomputable instance decidableRestrictDegree (m : ℕ) :
     DecidablePred (· ∈ { n : σ →₀ ℕ | ∀ i, n i ≤ m }) := by
+  classical
```

Comment:
This one looks wrong: if `classical` is allowed, there is no need to unfold anything, and the instance doesn't make sense. Instead, could you add suitable decidability assumptions for the instance?

Reply:
Good catch! I'm not sure what a suitable decidability assumption is... I have used `open Classical in` for now (which is less wrong, but perhaps not "right" yet).